zerver/actions/realm_settings.py

```diff
@@ -442,12 +442,21 @@ def do_change_realm_plan_type(
         realm.max_invites = settings.INVITES_DEFAULT_REALM_DAILY_MAX
         realm.message_visibility_limit = Realm.MESSAGE_VISIBILITY_LIMITED
         realm.upload_quota_gb = Realm.UPLOAD_QUOTA_LIMITED
+        # We do not allow public access on limited plans.
+        realm.enable_spectator_access = False
     else:
         raise AssertionError("Invalid plan type")
 
     update_first_visible_message_id(realm)
 
-    realm.save(update_fields=["_max_invites", "message_visibility_limit", "upload_quota_gb"])
+    realm.save(
+        update_fields=[
+            "_max_invites",
+            "enable_spectator_access",
+            "message_visibility_limit",
+            "upload_quota_gb",
+        ]
+    )
```

Comment:
I think if we wanted to do that, the right mechanism would be to call `do_set_realm_property`; which might be a good way to implement this from an abstractions standpoint anyway. I think a coherent way to do it would be to call that from the top of `do_change_plan_type`, before we do the other changes, basically starting with:
```
if plan_type == Realm.PLAN_TYPE_LIMITED and realm.enable_spectator_access:
     # comment
     do_set_realm_property(...)
```